### PR TITLE
graph: getGradient should always return a string

### LINKED ISSUE #2900: Graphs interactions failures
### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py version 4725c70c7ed724e2d1b9ba5618d7c30b957ee8a4

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=7, micro=4, releaselevel='final', serial=0)
INFO: os.name: posix
INFO: os.uname(): posix.uname_result(sysname='Linux', nodename='[...]', release='5.3.8-arch1-1', version='#1 SMP PREEMPT @1572357769', machine='x86_64')
INFO: sys.getwindowsversion(): N/A

--- check: package_management
INFO: has conda-meta: False
INFO: $VIRTUAL_ENV: '/home/[...]/venv'

--- check: installed_packages
INFO: installed: tensorboard==2.0.1
INFO: installed: tensorflow-gpu==2.0.0
INFO: installed: tensorflow-estimator==2.0.1

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.0.1'

--- check: tensorflow_python_version
INFO: tensorflow.__version__: '2.0.0'
INFO: tensorflow.__git_version__: 'v2.0.0-rc2-26-g64c3d38'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'/home/[...]/venv/bin/tensorboard\n'

--- check: readable_fqdn
INFO: socket.getfqdn(): 'localhost'

--- check: stat_tensorboardinfo
INFO: directory: /tmp/.tensorboard-info
INFO: os.stat(...): os.stat_result(st_mode=16895, st_ino=461902, st_dev=45, st_nlink=2, st_uid=1000, st_gid=1000, st_size=60, st_atime=1573120987, st_mtime=1573140549, st_ctime=1573140549)
INFO: mode: 0o40777

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['/home/[...]/venv/lib/python3.7/site-packages']; bad_roots (0): []

--- check: full_pip_freeze
INFO: pip freeze --all:
absl-py==0.8.0
astor==0.8.0
atomicwrites==1.3.0
attrs==19.1.0
cachetools==3.1.1
certifi==2019.6.16
chardet==3.0.4
coverage==4.5.4
cycler==0.10.0
gast==0.2.2
google-auth==1.7.0
google-auth-oauthlib==0.4.1
google-pasta==0.1.7
graphviz==0.13
grpcio==1.25.0
h5py==2.10.0
idna==2.8
importlib-metadata==0.20
Keras-Applications==1.0.8
Keras-Preprocessing==1.1.0
kiwisolver==1.1.0
Markdown==3.1.1
matplotlib==3.1.1
more-itertools==7.2.0
numpy==1.17.2
oauthlib==3.1.0
opt-einsum==3.0.1
packaging==19.1
pandas==0.25.1
pip==19.3.1
plotly==4.2.1
pluggy==0.12.0
progress==1.5
protobuf==3.9.1
psutil==5.6.3
py==1.8.0
pyarrow==0.14.1
pyasn1==0.4.7
pyasn1-modules==0.2.7
pydot==1.4.1
pyparsing==2.4.2
pytest==5.1.2
pytest-cov==2.7.1
pytest-timeout==1.3.3
python-dateutil==2.8.0
pytz==2019.2
requests==2.22.0
requests-oauthlib==1.3.0
retrying==1.3.3
rsa==4.0
scipy==1.3.1
seaborn==0.9.0
setuptools==41.2.0
six==1.12.0
tensorboard==2.0.1
tensorflow-estimator==2.0.1
tensorflow-gpu==2.0.0
termcolor==1.1.0
urllib3==1.25.3
wcwidth==0.1.7
Werkzeug==0.15.6
wheel==0.33.6
wrapt==1.11.2
zipp==0.6.0

``````

</details>

Browser informations:
``````
Chromium
Version 78.0.3904.97 (Build officiel) Arch Linux (64 bits)
``````

## Issue description

When a graph is displayed in the "GRAPHS" tag of tensorboard and the "Color" selector is set on "Structure":
- Node colors are relevant
- I can expand a node double-clicking on it
- The top right panel display informations about the currently selected node

When I change the "Color" selector (tested with "Device" and "TPU Compatibility"):
- Node color doesn't update until I hover the node with my mouse (look at "dense" node).
- When I hover the node, it turn gray, not respecting the color legend (red and green for TPU compatibility) (see dense_2, dense_3, loss, metrics).
- Expanding a node make the arrow move but not the nodes (below, the "dense_1" node is expanded).
- Selecting another node doesn't update the top right panel (below, I selected the "dense_3" node, but the panel still show "Loss" informations)

image

Switching back to the "Structure" colors, make the graph behavior back to the normal.

### PR DESCRIPTION
**Pull request by psybuzz**
Fixes #2900

* Motivation for features / changes
#2900
Several places in `tensorboard/plugins/graph/tf_graph_common/node.ts` use `getGradient()`, expecting it to return a string.  The function doesn't always return a string, which causes errors in the Graph dashboard.

* Technical description of changes
This PR ensures that `getGradient()` always returns a string.

* Detailed steps to verify changes work correctly (as executed by you)
Retried the repro steps for #2900 and saw that it now works correctly (graph is not visually broken, console errors from before are gone).
I haven't written a test, but given the complexity of GraphV1 and contextual knowledge required, I'd like to know whether it makes sense to write one now, or just wait for GraphV2.

### CODE REVIEW

**Comment by psybuzz** [comment]
Thanks for the review!

It does not fix the original issue mentioned.  However, I think that behavior is actually intended.  The TPU compatibility pane persists, even when no node is selected, because it shows the overall graph-wide compatibility.

**Comment by davidsoergel** [comment]
Awesome, thanks for doing this!  Post-hoc LGTM :)
